tests/unit/model-test.js

```diff
@@ -164,6 +164,82 @@ test("isDirty on the individual property is reset after save", function(assert){
     assert.equal(undefined, brandon.get("firstNameIsDirty"));
 });
 
+test("isDirty on the model is reset only after all values set back to original values", function(assert){
+    brandon = Person.create(data);
+    assert.equal("Brandon", brandon.get("firstName"));
+    assert.equal("Williams", brandon.get("lastName"));
+    assert.equal(false, brandon.get("isDirty"));
+    brandon.set("firstName", "foo");
+    brandon.set("lastName", "bar");
+    assert.equal(true, brandon.get("isDirty"));
+    brandon.set("firstName", "Brandon");
+    assert.equal(true, brandon.get("isDirty"));
+    brandon.set("lastName", "Williams");
+    assert.equal(false, brandon.get("isDirty"));
+});
+
+test("isDirty on the model is reset after value set back to original value", function(assert){
+    brandon = Person.create(data);
+    assert.equal("Brandon", brandon.get("firstName"));
+    assert.equal(false, brandon.get("isDirty"));
+    brandon.set("firstName", "baz");
+    assert.equal(true, brandon.get("isDirty"));
+    brandon.set("firstName", "Brandon");
+    assert.equal(false, brandon.get("isDirty"));
+});
+
+test("isDirty on the model is reset when original value is empty string and set back to empty string", function(assert){
+    brandon = Person.create({id: 1, firstName: "", lastName: "Williams"});
+    assert.equal(undefined, brandon.get("firstName"));
+    assert.equal(false, brandon.get("isDirty"));
```

Comment:
I was able to get this working correctly using this for the primed value

```js
var primed = Ember.isNone(value) && !data[key];
```

the problem is that if I add this the validation user experience is completely broken :( because when the user clicks into an input it immediately changes the model state to "" (empty string) causing the validation to fire right away. I'm thinking of a release where if we add this updated logic we could update the component to be smarter about "isDirty" (as this literally changes is the isDirty value on the model)